cpu/gd32v/periph/gpio.c

```diff
@@ -184,4 +205,98 @@ void gpio_write(gpio_t pin, int value)
     }
 }
 
+#ifdef MODULE_PERIPH_GPIO_IRQ
+
+/* Forward declaration of ISR */
+static void _gpio_isr(unsigned irqn);
+
+int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
+                  gpio_cb_t cb, void *arg)
+{
+    assert(cb != NULL);
+
+    int pin_num = _pin_num(pin);
+    int port_num = _port_num(pin);
+
+    /* disable interrupts on the channel we want to edit (just in case) */
+    EXTI->INTEN &= ~(1 << pin_num);
+
+    /* configure pin as input */
+    gpio_init(pin, mode);
+
+    /* GD32V has 16 EXTI lines for GPIO interrupts, where all ports share
+     * the same EXTI line for the same pin. That means the pin PA<n> shares
+     * the EXTI line with PB<n>, PC<n>, PD<n> and PE<n>. */
+    if (exti_ctx[pin_num].cb != 0) {
+        LOG_ERROR("EXTI line for GPIO_PIN(%u, %u) is already used.\n",
+                  port_num, pin_num);
```

Comment:
I guess we can't detect which one it was inside the ISR?

Better add an `assert(0)` here as config errors should be caught early.

Reply:
Initially I had an assert here, but since all other implementations don't check it, I replaced it with an error message.

Reply:
We could detect which one already occupies the EXTI line by holding a `gpio_t exti_used_by[16]` array which would result into 48 extra bytes. I'm not sure whether this would be worth. We could use a compress port/pin num format so that 1 byte per EXTI line would be sufficient since the GD32VF103 has 5 ports at maximum and 16 pins each port.

Reply:
Ah just for the debug output it's not needed.
I thought maybe there is a way to multiplex the interrupt itself, but probably not.

Reply:
Unfortunatly not.